release-process.md

```diff
@@ -178,38 +169,6 @@ Publishing to CRAN is done using <a href="https://cran.r-project.org/submit.html
 Since it requires further manual steps, please also contact the <a href="mailto:private@spark.apache.org">PMC</a>.
 
 
-<h4> Remove RC artifacts from repositories</h4>
```

Comment:
I moved the cleanup process to the end, since the docs are still needed when updating spark-website